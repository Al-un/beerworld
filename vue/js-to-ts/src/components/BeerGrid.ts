import { Component, Prop, Vue } from "vue-property-decorator";

import { Beer } from "@/models/beers";
import BeerGridItem from "./BeerGridItem.vue";

@Component({ components: { BeerGridItem } })
export default class BeerGrid extends Vue {
  @Prop({ type: Array, required: true })
  readonly beers!: Beer[];

  @Prop({ type: Boolean, default: false })
  readonly loading!: boolean;

  get beersCount(): number {
    return this.beers.length;
  }
}
