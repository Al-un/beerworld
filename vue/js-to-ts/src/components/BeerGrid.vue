<template>
  <div class="beer-grid">
    <p v-show="loading">Loading...</p>
    <beer-grid-item v-for="(beer, idx) in beers" :key="idx" :beer="beer" />
  </div>
</template>

<script lang="ts" src="./BeerGrid.ts"></script>

<style lang="scss">
.beer-grid {
  display: grid;
  gap: 16px;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  margin: auto;
  max-width: 700px;
  width: 100%;
}
</style>
