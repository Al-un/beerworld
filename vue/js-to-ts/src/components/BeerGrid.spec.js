import { mount } from "@vue/test-utils";

import BeerGrid from "./BeerGrid";

let wrapper;

const mockBeers = [
  { name: "beer1", mainPicture: "mainPic1" },
  { name: "beer2", mainPicture: "mainPic2" },
];

describe("components/BeerGrid", () => {
  it.each([
    ["When loading", true, mockBeers],
    ["When not loading", false, mockBeers],
    ["When beers list is empty", false, []],
  ])("renders and matches snapshot %s", (_, loading, beers) => {
    wrapper = mount(BeerGrid, {
      propsData: { beers, loading },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
