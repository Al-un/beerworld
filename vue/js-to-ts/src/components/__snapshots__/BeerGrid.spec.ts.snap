// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BeerGrid renders and matches snapshot When beers list is empty 1`] = `
<div class="beer-grid">
  <p style="display: none;">Loading...</p>
</div>
`;

exports[`components/BeerGrid renders and matches snapshot When loading 1`] = `
<div class="beer-grid">
  <p>Loading...</p>
  <div class="beer-grid-item"><img src="mainPic1" class="beer-img">
    <div class="beer-name">beer1</div>
  </div>
  <div class="beer-grid-item"><img src="mainPic2" class="beer-img">
    <div class="beer-name">beer2</div>
  </div>
</div>
`;

exports[`components/BeerGrid renders and matches snapshot When not loading 1`] = `
<div class="beer-grid">
  <p style="display: none;">Loading...</p>
  <div class="beer-grid-item"><img src="mainPic1" class="beer-img">
    <div class="beer-name">beer1</div>
  </div>
  <div class="beer-grid-item"><img src="mainPic2" class="beer-img">
    <div class="beer-name">beer2</div>
  </div>
</div>
`;
